egs/wsj/s5/local/nnet3/run_discriminative.sh

```diff
@@ -0,0 +1,224 @@
+#!/bin/bash
+
+set -e 
+set -o pipefail
+
+# this is run_discriminative.sh
+
+# This script does discriminative training on top of nnet3 system.
+# note: this relies on having a cluster that has plenty of CPUs as well as GPUs,
+# since the lattice generation runs in about real-time, so takes of the order of
+# 1000 hours of CPU time.
+# 
+# Note: rather than using any features we have dumped on disk, this script
+# regenerates them from the wav data three times-- when we do lattice
+# generation, numerator alignment and discriminative training.  This made the
+# script easier to write and more generic, because we don't have to know where
+# the features and the iVectors are, but of course it's a little inefficient.
+# The time taken is dominated by the lattice generation anyway, so this isn't
+# a huge deal.
+
+. cmd.sh
+
+
+stage=0
+train_stage=-10
+get_egs_stage=-10
+use_gpu=true
+srcdir=exp/nnet3/nnet_ms_a
+criterion=smbr
+drop_frames=false  # only matters for MMI.
+frames_per_eg=150
+frames_overlap_per_eg=30
+effective_learning_rate=0.0000125
+num_jobs_nnet=4
+train_stage=-10 # can be used to start training in the middle.
+decode_start_epoch=1 # can be used to avoid decoding all epochs, e.g. if we decided to run more.
+num_epochs=4
+degs_dir=
+cleanup=false  # run with --cleanup true --stage 6 to clean up (remove large things like denlats,
+               # alignments and degs).
+lats_dir=
+train_data_dir=data/train_si284_hires
+online_ivector_dir=exp/nnet3/ivectors_train_si284
+one_silence_class=true
+truncate_deriv_weights=10
+minibatch_size=64
+
+adjust_priors=true
+
+determinize=true
```

Comment:
Vimal, I think now that you've worked out the recipe, this whole block of configuration values should be removed and just hardcoded.  They're not the kind of things we expect users will want to play with.